src/Diagnostics.DataProviders/ConfigurationFactory.cs

```diff
@@ -24,6 +27,17 @@ public AppSettingsDataProviderConfigurationFactory(IHostingEnvironment env)
                 .SetBasePath(Directory.GetCurrentDirectory())
                 .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: true);
 
+            var tokenProvider = new AzureServiceTokenProvider();
+            var keyVaultClient = new KeyVaultClient(
+                new KeyVaultClient.AuthenticationCallback(
+                    tokenProvider.KeyVaultTokenCallback
```

Comment:
Will this open up a modal box to sign in?

Reply:
yes

Reply:
How will this run in Production then?

Reply:
there are multiple ways to get credentials: this initial change is to allow key vault for dev access, I intend to build on top of this to allow keyvault for production

Reply:
This uses multiple methods of authentication. one of which is a modal box another is visual studio credentials and in azure you can use the resource ID.